Code formatting setup and CI

danpalmer opened the pull request:
As discussed in #133, this PR aims to introduce a basic code style, auto formatting setup, and linting for common Python coding errors.

The diff here is huge, but most of it comes from one commit that was almost entirely automated and can therefore probably be mostly ignored. I recommend reviewing the rest by commit. I've included some more background in commit messages where I can.

The changes here do fix some potential bugs, and I hope improve the consistency of some aspects of the code.

I'm open to discussion about the sorts of things being linted here, I realise this is quite a big change to the developer workflow and I'm keen to take into account existing style within Monzo for example. This setup is a simple version of what we use at Thread, and we find it to be effective in helping us to maintain a high code quality.

milesbxf commented:
Thanks so much for this - unfortunately only saw it after we merged #134 so I'm afraid I've introduced conflicts already 😭 

I'll make sure we don't merge anything else in the meantime - let me know once you've rebased and I'll prioritise looking at this and getting it merged!

danpalmer commented:
@milesbxf Hey, I think I've addressed all the merge conflicts. I also found another flaky test and fixed that. If it's at all possible could you have a look at this today and perhaps cut another release to PyPI? I'm keen to work on some tooling using this at the weekend, and may want to contribute more changes here depending on how that goes. It would be great to be able to do this on top of these changes.

milesbxf commented on response/slack/client.py:
```diff
@@ -72,7 +68,7 @@ def get_channel_id(self, name):
                 next_cursor = response["response_metadata"]["next_cursor"]
                 logger.info(f"get_channel_id - next_cursor == [{next_cursor}]")
-            except:
+            except LookupError:
```
Just checking - was catching this specific exception intentional?

Don't think it matters, the error handling here isn't great anyway to be honest

danpalmer replied:
Yes. "Blind excepts" are banned under the linter to force more careful exception handling. In this case my assumption was that the error was missing details in the `response` dict, so I went with a `LookupError` as a fairly generic thing to catch.

danpalmer commented:
@milesbxf no problem, happy to do it, keen to get us using response internally too.

danpalmer commented:
Thanks for the speedy review, much appreciated!
